build_tools/docker/build_and_update_gcr.py

```diff
@@ -89,6 +89,7 @@ def parse_arguments():
     images_to_update.extend(IMAGES_TO_DEPENDENT_IMAGES[args.image])
 
   for image in images_to_update:
+    print(f"Updating image {image}")
```

Comment:
```suggestion
    print(f'Updating the {image} image.')
```
nit: File quote consistency / wording.

I think something like this would also be good for debugging and bring the output here more in line with `set -x`

```python3
    build_cmd = ['docker', 'build', '--tag', image_url, image_path]
    print(' '.join(build_cmd))
    subprocess.check_output(build_cmd)
    if args.push:
      push_cmd = ['docker', 'push', image_url]
      print(' '.join(push_cmd)
      subprocess.check_output(push_cmd)
```

Reply:
Fixed the quoting. Better debug logging SGTM, but let's do it separately. We might also want to print the subprocess output so people can see what's going on. I think that will require communicate. This is the advantage of using shell scripts :-D